llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -217,6 +217,74 @@ bool AMDGPUInstructionSelector::selectCOPY(MachineInstr &I) const {
   return true;
 }
 
+bool AMDGPUInstructionSelector::selectCOPY_SCC_VCC(MachineInstr &I) const {
```

Comment:
I'm wondering if we should just avoid this situation

Reply:
This allows for more registers to be allocated to sgpr.
Avoiding them could be done later in some optimization pass when we can decide if we want to do global-isel equivalent of moveToVALU, but we don't do it for correctness but to influence register allocation to sgpr or vgpr.
Why would we want to avoid vcc to scc copy? Question for @nhaehnle also.